src/Sentry.Extensions.Logging/SentryStructuredLogger.cs

```diff
@@ -0,0 +1,121 @@
+using Microsoft.Extensions.Logging;
+using Sentry.Extensibility;
+using Sentry.Infrastructure;
+
+namespace Sentry.Extensions.Logging;
+
+internal sealed class SentryStructuredLogger : ILogger
+{
+    private readonly string? _categoryName;
+    private readonly SentryLoggingOptions _options;
+    private readonly IHub _hub;
+    private readonly ISystemClock _clock;
+    private readonly SdkVersion _sdk;
+
+    internal SentryStructuredLogger(string categoryName, SentryLoggingOptions options, IHub hub, ISystemClock clock, SdkVersion sdk)
+    {
+        _categoryName = categoryName;
+        _options = options;
+        _clock = clock;
+        _hub = hub;
+        _sdk = sdk;
+    }
+
+    public IDisposable? BeginScope<TState>(TState state) where TState : notnull
+    {
+        return NullDisposable.Instance;
```

Comment:
What's our plan for this then? Do we just not support BeginScope for structured logging? Or are we relying on the scope having been pushed somewhere else?